### DISCUSSION TITLE
How to inject the AST location

### DISCUSSION THREAD

**Discussion by machaval** [kevinmehall/rust-peg#346] [General]
I'm writing a parser and I have as a requirement that the AST has the location of the node in the text.
I'm not able to find an example that does this

**Comment by kevinmehall** [comment]
`position!()` matches the empty string and returns the current offset into the input. See https://github.com/signalspec/signalspec/blob/master/src/syntax/grammar.rs for a complete example of using it to attach spans to AST nodes.
